File: python/day02_2.py
from day02_1 import HI, LETTER, LO, PWD, count_valid
from file_ops import readlines


def is_valid(groupdict):
    first = int(groupdict[LO]) - 1
    second = int(groupdict[HI]) - 1
    letter = groupdict[LETTER]
    password = groupdict[PWD]

    return (password[first] == letter) != (password[second] == letter)


if __name__ == '__main__':
    lines = readlines(2)
    print(count_valid(lines, is_valid))

File: node/day02_2.js
const fs = require('fs');

const data = fs.readFileSync('../data/input02.txt').toString('utf-8');
const lines = data.split('\n');

const valid = [];

function isValid(line) {
    if (!line) {
        return;
    }

    const password = line.split(':')[1].trim();
    const first = parseInt(line.split('-')[0]) - 1;
    const second = parseInt(line.split('-')[1].split(' ')[0]) - 1;
    const char = line.split(':')[0].split(' ')[1];

    if ((password[first] === char) !== (password[second] === char)) {
        valid.push(1);
    }
}

lines.forEach(isValid);
console.log(valid.length);
